internal/server/models/provideruser.go

```diff
@@ -12,7 +12,6 @@ type ProviderUser struct {
 	ProviderID uid.ID `gorm:"primaryKey"`
 
 	Email      string
-	Groups     CommaSeparatedStrings
```

Comment:
So where is the provider_groups_users table now?

Reply:
It's this relation table:
```
CREATE TABLE provider_groups_identities (
    provider_group_id bigint NOT NULL,
    identity_id bigint NOT NULL
);
```

It doesn't tie directly to the provider user so that we don't have to do an extra join on membership look up.

Reply:
I'm not a fan of the membership lookup model you've selected here. Instead we should allow sync to modify our group membership table and keep the queries simple; leave the complication in the sync and don't pay for it throughout the app.

Reply:
Resolved the complexity in the new change by using a table view that makes queries simple (see the schema.sql). I tried resolving membership with application logic in this branch: https://github.com/infrahq/infra/commit/82a2adbea3c3412d45050b1ad5ccbc715ce4613a and it seems more error prone than the `VIEW` approach which makes managing group membership and resolving group membership decently understandable.